src/AppInstallerSharedLib/Versions.cpp

```diff
@@ -64,7 +64,20 @@ namespace AppInstaller::Utility
             size_t newPos = baseVersion.find_first_of(splitChars, pos);
 
             size_t length = (newPos == std::string::npos ? baseVersion.length() : newPos) - pos;
-            m_parts.emplace_back(baseVersion.substr(pos, length));
+            std::string interimPart = baseVersion.substr(pos, length);
+
+            // For the first version part, if there is a digit before the split character, trim off all leading non-digit characters

```

Comment:
This seems like it would be better done by modifying `baseVersion` before the loop rather than having a conditional that we know will only ever be hit once.